todo_app/todo_view_model.py

```diff
@@ -0,0 +1,74 @@
+class ToDoViewModel:
+    def __init__(self, items, recent_date):
+        self._items = items
+        self.determine_done_items() 
+        self.determine_recent_done_items(recent_date)
+        self.determine_older_done_items(recent_date)
+        self.determine_show_all_done()
+
+    @property
+    def items(self):
+        return self._items
+
+    @property
+    def to_do_items(self):
+        todo_items=[]
+        for item in self._items:
+            if item.status.upper() == "TO DO":
+                todo_items.append(item)
+        return todo_items
+
+    @property
+    def doing_items(self):
+        doing_items=[]
+        for item in self._items:
+            if item.status.upper() == "DOING":
+                doing_items.append(item)
+        return doing_items
+
+    @property
+    def done_items(self):
+        return self._done_items
+        
+    @property
+    def show_all_done(self):
+        return self._show_all_done
+    
+    @show_all_done.setter
+    def show_all_done(self, value):
+        self._show_all_done = value
+
+    @property 
+    def recent_done_items(self):
+        return self._recent_items
+
+    @property
+    def older_done_items(self):
+        return self._older_items
+
+    def determine_show_all_done(self):
+        done_items = self.done_items
+        if len(done_items) >= 5:
+            self._show_all_done = False
```

Comment:
This property doesn't really belong to this class. I'd just pass it through to the `render_template` method and use it in the template. Should allow you strip out alot of methods here and reduces the number of tests.